pkg/osbuild/bootc_install_to_filesystem_stage_test.go

```diff
@@ -0,0 +1,82 @@
+package osbuild_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/images/pkg/osbuild"
+)
+
+func TestBootcInstallToFilesystemStageNewHappy(t *testing.T) {
+	devices := makeOsbuildDevices("dev-/", "dev-/boot", "dev-/boot/efi")
+	mounts := makeOsbuildMounts("/", "/boot", "/boot/efi")
+
+	expectedStage := &osbuild.Stage{
+		Type:    "org.osbuild.bootc.install-to-filesystem",
+		Devices: devices,
+		Mounts:  mounts,
+	}
+	stage, err := osbuild.NewBootcInstallToFilesystemStage(devices, mounts)
+	require.Nil(t, err)
+	assert.Equal(t, stage, expectedStage)
+}
+
+func TestBootcInstallToFilesystemStageMissingMounts(t *testing.T) {
+	devices := makeOsbuildDevices("dev-/")
+	mounts := makeOsbuildMounts("/")
+
+	stage, err := osbuild.NewBootcInstallToFilesystemStage(devices, mounts)
+	// XXX: rename error
+	assert.ErrorContains(t, err, "required mounts for bootupd stage [/boot /boot/efi] missing")
+	require.Nil(t, stage)
+}
+
+func TestBootcInstallToFilesystemStageJsonHappy(t *testing.T) {
+	devices := makeOsbuildDevices("disk", "dev-/", "dev-/boot", "dev-/boot/efi")
+	mounts := makeOsbuildMounts("/", "/boot", "/boot/efi")
+
+	stage, err := osbuild.NewBootcInstallToFilesystemStage(devices, mounts)
+	require.Nil(t, err)
+	stageJson, err := json.MarshalIndent(stage, "", "  ")
+	require.Nil(t, err)
+	assert.Equal(t, string(stageJson), `{
+  "type": "org.osbuild.bootc.install-to-filesystem",
+  "devices": {
+    "dev-/": {
+      "type": "orgosbuild.loopback"
+    },
+    "dev-/boot": {
+      "type": "orgosbuild.loopback"
+    },
+    "dev-/boot/efi": {
+      "type": "orgosbuild.loopback"
```

Comment:
orgosbuild!  It sounds like a battle cry a barbarian warrior might make to get their computer to work or something.

Reply:
I was wondering how the test was passing and now realise that this is a pre-existing typo from 159551483f which I missed!

Reply:
*cough* I will grab a brown paperbag. Sorry for that! Bug is fixed in my next push.